### ISSUE TITLE
submission unlinked from project

### ISSUE THREAD

**Issue by amnonkhen** [ebi-ait/dcp-ingest-central#612]
**Describe the bug**
discovered by @ipediez

**To Reproduce**
Steps to reproduce the behaviour:
1. start with an existing project - autoimported
2. update project info
3. created a submission by uploading spreadsheet
4. notice submission is not linked to project

**Expected behaviour**
Submission should be linked to project

**Screenshots / Logs**
> If applicable, add screenshots or log files to help explain your problem.

**Links**
> If applicable, link to relevant docs or Slack conversations.

**Environment**
- [x] prod
- [ ] staging
- [ ] dev

**Browser**
- [ ] Chrome
- [x] Safari
- [ ] IE
- [ ] Other (please specify)

**Comment by ipediez** [comment]
### Project 1
https://contribute.data.humancellatlas.org/projects/detail?uuid=cd61771b-661a-4e19-b269-6e5d95350de6

**Submissions:**

1. https://contribute.data.humancellatlas.org/submissions/detail?uuid=7b1f1a8d-4aa6-4d2d-a675-5b0589ed2d65
This one, by 11:47 13.01.22 is still linked. Only the spreadsheet has been uploaded.

2. https://contribute.data.humancellatlas.org/submissions/detail?uuid=4bc044b2-aa53-4c8f-8d5e-a40e5b6c4cf1
In this one, the spreadsheet was uploaded and the files were also uploaded to the s3 bucket. All the metadata was valid, and after hitting the "Validate Submission" button for graph validation it showed the following error message and was unlinked, 

### Project 2

**Submissions** (older to newer):
https://contribute.data.humancellatlas.org/submissions/detail?uuid=1c82779c-6feb-4824-86c4-6648f8b63167
https://contribute.data.humancellatlas.org/submissions/detail?uuid=0b443736-0ac1-4e3c-aeef-ba9c0c7e1302
https://contribute.data.humancellatlas.org/submissions/detail?uuid=163719fe-ebbf-4b0f-a1f8-80198da44166
https://contribute.data.humancellatlas.org/submissions/detail?uuid=83644ebc-099e-4e96-90af-2201465235e4
https://contribute.data.humancellatlas.org/submissions/detail?uuid=1cb5a259-7f06-46b1-a33d-2b23467c0a79

In all the cases, the only step performed was to upload the spreadsheet to create a submission.

**Comment by idazucchi** [comment]
I think I might be facing the same problem, at least with one project
### Project 3
https://contribute.data.humancellatlas.org/projects/detail?uuid=d71c76d3-3670-4774-a9cf-034249d37c60&tab=upload

**Submissions**
Oldest to newest, in both cases I only uploaded the spreadsheet
1. https://contribute.data.humancellatlas.org/submissions/detail?uuid=981498d4-ff17-495d-9bb9-f9076d25fbf9
2. https://contribute.data.humancellatlas.org/submissions/detail?uuid=7e44319e-f382-4116-8b5d-9c9395643c8a

**Comment by ESapenaVentura** [comment]
@aaclan-ebi has a suspicion on what the error is, she will investigate (After #616)

**Comment by aaclan-ebi** [comment]
It seems that I've unintentionally changed the behaviour on when the submission will be linked to the project as part of my changes in ebi-ait/dcp-ingest-central#549
https://github.com/ebi-ait/ingest-client/compare/d5654ab..04e6d5e?diff=split#diff-2617875dbb190b4041551953a829be7e70dc9172d66c8b63df7fb46d079d6c71L77-L79

**Before**:  The linking of submission to project will happen AFTER **creating the new entities in the submission** and BEFORE **the linking of entities**
**Now**: The linking of submission to project will happen AFTER **creating the new entities** and **linking of entities**

However, this shouldn't cause the 401 errors that happened on the @ipediez's and @idazucchi 's submissions. 
It seems that this is an issue when the token expires during importing. There's an issue filed in 2020 related to this ebi-ait/dcp-ingest-central#549.  This can happen when
1.  spreadsheets are large
2. user uploads spreadsheet after being logged in for some time and token is about to expire

For project 1, submission 2 (with uuid `4bc044b2-aa53-4c8f-8d5e-a40e5b6c4cf1`) the submission is not that large, I suspect the spreadsheet was uploaded after some time of being logged in. 

@ipediez Do you want to delete project 1's submission 1 (with uuid `7b1f1a8d-4aa6-4d2d-a675-5b0589ed2d65`) and link submission 2 to that project ? I could do that so that you'll be able to submit that valid submission.

It seems that all other submissions that Irene and Ida noted are bigger and have more linkings. We need to fix the issue ebi-ait/dcp-ingest-central#549 to unblock these submissions.

**Comment by ipediez** [comment]
@aaclan-ebi yes, it would be perfect if you could link that submission, which is already graph valid. Thanks!

**Comment by gabsie** [comment]
New ticket has been created but it seems the Project overwrite has changed the behaviour of the submission linking.
@amnonkhen and @aaclan-ebi will discuss after standup how to unblock projects exporting.

**Comment by aaclan-ebi** [comment]
1 - [7b1f1a8d-4aa6-4d2d-a675-5b0589ed2d65](https://contribute.data.humancellatlas.org/submissions/detail?uuid=7b1f1a8d-4aa6-4d2d-a675-5b0589ed2d65) total count of entities 722
error: no error, this submission is linked to project HumanColonRewiringUlcerativeColitis and needs to be deleted

2 - [4bc044b2-aa53-4c8f-8d5e-a40e5b6c4cf1](https://contribute.data.humancellatlas.org/submissions/detail?uuid=4bc044b2-aa53-4c8f-8d5e-a40e5b6c4cf1) , total count: 722
error: failed in linking submission to project because of expired token
```
Received an HTTP 401 from http://172.20.134.175/submissionEnvelopes/61dff94e7701fb51a5c2b974: 1642068814473401UnauthorizedUnauthorized/submissionEnvelopes/61dff94e7701fb51a5c2b974
```
action:
I’ll manually link the submission to project HumanColonRewiringUlcerativeColitis

3 - [1c82779c-6feb-4824-86c4-6648f8b63167](https://contribute.data.humancellatlas.org/submissions/detail?uuid=1c82779c-6feb-4824-86c4-6648f8b63167), total count of entities : 98868
error:
```
Received an HTTP 401 from http://172.20.134.175/submissionEnvelopes/61c2fc5df358643c2eb2faf8: 1640173826113401UnauthorizedUnauthorized/submissionEnvelopes/61c2fc5df358643c2eb2faf8
```

4 - [0b443736-0ac1-4e3c-aeef-ba9c0c7e1302](https://contribute.data.humancellatlas.org/submissions/detail?uuid=0b443736-0ac1-4e3c-aeef-ba9c0c7e1302) , total count: 98868

No error but it didn't finish creating all entities nor linking them - submitted last dec 22 - so this might be the one breaking the broker pod


5 - [163719fe-ebbf-4b0f-a1f8-80198da44166](https://contribute.data.humancellatlas.org/submissions/detail?uuid=163719fe-ebbf-4b0f-a1f8-80198da44166) , total count: 98868
```
Received an HTTP 401 from http://172.20.134.175/submissionEnvelopes/61c5f00634592953c85e00db: 1640366582644401UnauthorizedUnauthorized/submissionEnvelopes/61c5f00634592953c85e00db
```

6 - [83644ebc-099e-4e96-90af-2201465235e4](https://contribute.data.humancellatlas.org/submissions/detail?uuid=83644ebc-099e-4e96-90af-2201465235e4) , total count: 1566

This error is related to issue in Rachel’s dataset , the fix was promoted jan 12, the submission was created jan 10 - this can be retried
```
Received an HTTP 500 from http://172.20.134.175/processes/61dc749bc2be115bdc97ba8a/protocols: http://172.20.134.175/processes/61dc749bc2be115bdc97ba8a/protocolsUnexpected server error
```
action: I believe we can retry uploading the spreadsheet

7 - [1cb5a259-7f06-46b1-a33d-2b23467c0a79](https://contribute.data.humancellatlas.org/submissions/detail?uuid=1cb5a259-7f06-46b1-a33d-2b23467c0a79) , total count: 98868
error: It created all entities but nothing was linked because of 401 error in finding the project uuid
```
Received an HTTP 401 from http://172.20.134.175:80/projects/search/findByUuid?uuid=165dea71-a95a-44e1-88cd-b2d9ad68bb1e: 1641983173925401UnauthorizedUnauthorized/projects/search/findByUuid
```

8 - [981498d4-ff17-495d-9bb9-f9076d25fbf9](https://contribute.data.humancellatlas.org/submissions/detail?uuid=981498d4-ff17-495d-9bb9-f9076d25fbf9)
error: failed in linking submission to project
action: this has invalid protocol and a duplicate of submission 9, delete this and keep submission 9

9 - [7e44319e-f382-4116-8b5d-9c9395643c8a](https://contribute.data.humancellatlas.org/submissions/detail?uuid=7e44319e-f382-4116-8b5d-9c9395643c8a) 
error: failed in linking submission to project
action: link this submission to its project

**Comment by aaclan-ebi** [comment]
To summarise, there are 4 unique submissions noted above that are blocked:
- submission 2
- submission 9 
- submission 6 - owned by UCSC
- submission with 98868 entities 

Unblocked @ipediez and @idazucchi on their submissions

submission 2 - https://contribute.data.humancellatlas.org/projects/detail?uuid=cd61771b-661a-4e19-b269-6e5d95350de6&tab=upload

submission 9 - https://contribute.data.humancellatlas.org/projects/detail?uuid=d71c76d3-3670-4774-a9cf-034249d37c60&tab=upload

**Comment by aaclan-ebi** [comment]
**Dev notes:**

commands ran manually to link submission to project and delete submission errors:

```
$ TOKEN='insert-jwt-token-from-ui'
# link submission to project
$ curl -X POST -H "Authorization: Bearer $TOKEN" -H "Content-Type: text/uri-list" -d https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/<insert-submission-id> https://api.ingest.archive.data.humancellatlas.org/projects/<insert-project-id>/submissionEnvelopes
# delete submission errors
$ curl -X DELETE -H "Authorization: Bearer $TOKEN" https://api.ingest.archive.data.humancellatlas.org/submissionEnvelopes/<insert-submission-id>/submissionErrors
```

**Comment by aaclan-ebi** [comment]
I'll work on the fix on the importer code next to make sure linking of submission to project happens first when importing.

**Comment by idazucchi** [comment]
PR ready, review needed

**Comment by aaclan-ebi** [comment]
Another submission was affected by the linking issue https://contribute.data.humancellatlas.org/submissions/detail?uuid=310e23bf-ed26-4447-a2d5-d666e558b3a1. The manual fix was done.

The fix is now deployed in prod.
